mantidproject/mantid
# Rewrite of Chop with new GUI, including MultiRep for LET.

This is a rewrite of PyChop with a new GUI and CLI. The PyChop code previously in Mantid (by JWT) was based on an incomplete Matlab port of the original Fortran code (by TGP), whilst this version includes a more faithful translation (by RAE) of the Fortran. In addition it includes code (by RIB and DJV) to calculate the multiple reps allowed by Merlin and LET, and the resolution of LET which CHOP does not support.

**To test:**

Run through the attached IPython notebook: 
[PyChop2_ipynb.zip](https://github.com/mantidproject/mantid/files/438300/PyChop2_ipynb.zip)

Run MantidPlot - there should be a new interface under `Direct` called `PyChop`. 

Or, in a MantidPython shell do:

```
from PyChop import PyChop2, PyChopGui
merlin = PyChop2('merlin','s',250)
merlin.setEi(25)
resolution = merlin.getResolution()
print resolution
resolution, flux = PyChop2.calculate(inst='maps', chtyp='a', freq=500, ei=600, etrans=range(0,550,50))
print resolution
print flux
PyChop2.showGUI()
```

Expected output:

```
[ 0.53847416  0.54998197  0.56540704  0.5843418   0.6065874   0.63201212
  0.66050558  0.691962    0.72627472  0.76333545  0.80303543  0.84526708
  0.88992562  0.93691037  0.98612563  1.03748123  1.09089277  1.14628162
  1.20357476]
[ 17.12380445  16.00789732  14.95785911  13.97730028  13.07002193
  12.23992828  11.49090381  10.82667552  10.25072764   9.76644397
   9.37794616]
2379.08526288
```

Fixes #10147 



---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
